lib/health_checks/sidekiq.rb

```diff
@@ -15,18 +16,21 @@ def sidekiq(config)
   class LivenessServer
     LIVENESS_PORT = 8080
 
-    def start
+    def start(mongo_databases, redis_configs)
       Sidekiq::Logging.logger.info "Starting liveness server on #{LIVENESS_PORT}"
+
       Thread.start do
+        clients = create_clients(mongo_databases)
         server = TCPServer.new(LIVENESS_PORT)
         loop do
           Thread.start(server.accept) do |socket|
             begin
               HealthChecks.memory
-              check_redis
-              check_mongo
+              check_mongoid_clients(clients)
```

Comment:
You're not really checking that clients work, clients always work, you're checking the connectivity to Mongo hosts, same way as you are for Redis so name this consistently with the latter - `check_mongo_connections`